### ISSUE TITLE
Do not lock the HDF5 file with silx view

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#2859]
This would allow to open a file while another process (e.g., acquisition) is writing it.

**Comment by woutdenolf** [comment]
This could be an option (lock pool depends on what you're doing: threading, gevent, asyncio, ...)
```python
class File(h5py.File):
    def __init__(self, name, mode="r", enable_file_locking=None, creationlocks=None, **kwargs
    ):
        self._lockname = os.path.abspath(name)
        self._creationlocks = creationlocks
        with self.acquire_lock():
            if enable_file_locking is None:
                enable_file_locking = mode != "r"
            if enable_file_locking:
                os.environ["HDF5_USE_FILE_LOCKING"] = "TRUE"
            else:
                os.environ["HDF5_USE_FILE_LOCKING"] = "FALSE"
            super().__init__(name, mode=mode, **kwargs)

    @contextmanager
    def acquire_lock(self):
        if self._creationlocks is None:
            yield
        else:
            with self._creationlocks.acquire(self._lockname):
                yield
```

**Comment by t20100** [comment]
I made it simpler, at least for now: I set `HDF5_USE_FILE_LOCKING` when starting `silx view`.

What you propose would be best integrated into `h5py` directly.
BTW there is no other way than the environment variable?

**Comment by woutdenolf** [comment]
Yep no other way. Dirty but very necessary.

**Comment by t20100** [comment]
closed by #2859

**Comment by t20100** [comment]
For the record:
With h5py >= 2.10, one way to avoid the lock is to give h5py a file object rather than the file name:

```
f = h5py.File(open("myfile.h5", 'rb'), mode='r')
```
Opening it this way does not lock the file, so you can open it in append mode from another process.

Thanks to @kif for the idea
